kvdb/etcd/stm.go

```diff
@@ -104,15 +110,26 @@ func (e DatabaseError) Error() string {
 	return fmt.Sprintf("etcd error: %v - %v", e.msg, e.err)
 }
 
-// stmGet is the result of a read operation,
-// a value and the mod revision of the key/value.
+// stmGet is the result of a read operation, a value and the mod revision of the
+// key/value.
 type stmGet struct {
-	val string
+	KV
 	rev int64
 }
 
+// Less implements less operator for btree.BTree.
+func (c *stmGet) Less(than btree.Item) bool {
+	return c.key < than.(*stmGet).key
+}
+
 // readSet stores all reads done in an STM.
-type readSet map[string]stmGet
+type readSet struct {
+	// tree stores the items in the read set.
+	tree *btree.BTree
+
+	// fullRanges stores full range prefixes.
+	fullRanges map[string]bool
```

Comment:
`bool` has some semantics here, or a `struct{}` will do? Latter consumes less memory in the map (there's a singleton `struct{}` instance in the runtime).